kiosk/src/Components/AddGameButton.tsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from "react";
+import { Kiosk } from "../Models/Kiosk";
+import "../Kiosk.css";
+import { GameData } from "../Models/GameData";
+
+interface IProps {
+    kiosk: Kiosk;
+    selected: Boolean;
+    content: String;
+  }
+
+const AddGameButton: React.FC<IProps> = ({ kiosk, selected, content }) => {
+    const [buttonActive, setButtonState] = useState(selected);
+    let buttonColor = buttonActive ? "lightblue" : "transparent";
+
+    const updateButton = () => {
+        buttonColor = buttonActive ? "lightblue" : "transparent";
+    }
+
+    useEffect(() => {
+        setButtonState(selected);
+        updateButton();
+    });
```

Comment:
I think buttonActive isn't doing anything here / can just use selected directly and remove updateButton / this useEffect?

Reply:
It's possible that I could use something else to listen to the change in `selected`, or I might need to change how I communicate that the button is selected, but I need this `useEffect` due to how everything fits together currently.  The change in `selected` is being passed down from the main menu because the main menu has a game pad loop that listens for when a user "selects up". So, the rendering of the button itself will not be sufficient to update the button.

I need this `buttonSelected` state in the Main Menu to be passed to both the button and the game list to freeze the game pad loop that exists in the game list. If I don't have the controls frozen on the game list, when the user tries to trigger the button when it is in the selected state, a game will be launched instead because of the game pad loop in the game list. 

I figure there is a better way to do this by just having the game pad controls be fully in the main menu, but the loop in `GameList.tsx` is complicated enough that I figured that it would, at the time being, be better to leave the controls as they are.

Reply:
react will rerun this function when the prop changes (`selected`), and triggers an update to the actual dom if anything changes in the returned value -- in this case if buttonColor were determined by selected directly then it should be fine?